### ISSUE TITLE
No CLIENT_RANDOM output with Java 8, 11 and later

### ISSUE THREAD

**Issue by pirarucu** [neykov/extract-ssl-secrets#19]
In my tests, the callback function `onCalculateKeys` only seems to be called with Java 9 and 10. I could not get CLIENT_RANDOM with Java 8, 11 and later.

With Java 9 and 10, I can get both entries in the key log, e.g.,
```
RSA Session-ID: _xxx_ Master-Key: _yyy_
CLIENT_RANDOM _zzz_ _yyy_
```
But with Java 8, 11 and later, I'm only getting RSA Session-ID in the key log without CLIENT_RANDOM.

Not sure is anyone else experiencing this problem?

**Comment by neykov** [comment]
@pirarucu I can confirm the behaviour - I can replicate it on my side.
Is this preventing you from decrypting a session though? I am still able to decrypt with only "RSA Session-ID" available.

**Comment by pirarucu** [comment]
@neykov Thank you for confirming. It doesn't block the decryption. I just have a program to do some tracking with CLIENT_RANDOM keys. I worked around by this change.
_https://github.com/pirarucu/extract-tls-secrets/commit/693607cc757c0a43c797aa2c177552ea148f8b0f_

**Comment by neykov** [comment]
Thanks @pirarucu. Let me look into the changes to understand them better. Thanks for sharing.

**Comment by neykov** [comment]
@pirarucu I've pushed a change to master that should address the problem. Do you mind checking it out?
I'll push out a new release if it works for you.

**Comment by pirarucu** [comment]
Thanks @neykov. I'd love to check it out! I will get back to you later.

**Comment by pirarucu** [comment]
@neykov, I've tried the fix and it worked as expected. Many thanks for the help.

And after this change, it'd get multiple CLIENT_RANDOM keys with the same Master-Key when renegotiation happens. FYI.
> RSA Session-ID: _xxx_ Master-Key: _yyy_
> CLIENT_RANDOM _aaa_ _yyy_
> CLIENT_RANDOM _bbb_ _yyy_
> CLIENT_RANDOM _ccc_ _yyy_

**Comment by neykov** [comment]
The change is now available in the [5.0.0 release](https://github.com/neykov/extract-tls-secrets/releases/tag/v5.0.0).
